src/modules/sensors/voted_sensors_update.cpp

```diff
@@ -899,6 +899,45 @@ void VotedSensorsUpdate::baroPoll(vehicle_air_data_s &airdata)
 
 bool VotedSensorsUpdate::checkFailover(SensorData &sensor, const char *sensor_name, const uint64_t type)
 {
+	uint64_t type2 = type;
+
+	if (type == subsystem_info_s::SUBSYSTEM_TYPE_GYRO) { type2 = subsystem_info_s::SUBSYSTEM_TYPE_GYRO2; }
+
+	if (type == subsystem_info_s::SUBSYSTEM_TYPE_ACC) { type2 = subsystem_info_s::SUBSYSTEM_TYPE_ACC2; }
+
+	if (type == subsystem_info_s::SUBSYSTEM_TYPE_MAG) { type2 = subsystem_info_s::SUBSYSTEM_TYPE_MAG2; }
+
+	// set sensors health flags according to the sensor state
+	uint64_t curr_type;
+
+	for (uint8_t i = 0; i < sensor.subscription_count; i++) {
+		if (i == 0) {
+			curr_type = type;
+
+		} else if (i == 1) {
+			curr_type = type2;
+
+		} else {
+			break;
+		}
+
+		bool sensor_healthy = sensor.voter.get_sensor_state(i) == DataValidator::ERROR_FLAG_NO_ERROR;
+
+		uint64_t new_sensors_health = _sensors_health;
+
+		if (sensor_healthy) {
+			new_sensors_health |= curr_type;
+
+		} else {
+			new_sensors_health &= ~curr_type;
+		}
+
+		if (_sensors_health != new_sensors_health) {
```

Comment:
Maybe we should give commander some time to initialize and subscribe to subsystem_info since this is published on change thus failures might get dropped
for instance:
`if (_sensors_health != new_sensors_health && hrt_absolute_time > 10_s) {`